.cirrus.yml

```diff
@@ -22,584 +40,532 @@ task:
       env:
-        SHARD: integration_tests
-        SUBSHARD: gradle1
-      test_script:
-        # Unsetting CIRRUS_CHANGE_MESSAGE and CIRRUS_COMMIT_MESSAGE as they
-        # might include non-ASCII characters which makes Gradle crash.
-        # See: https://github.com/flutter/flutter/issues/24935
-        # This is a temporary workaround until we figure how to properly configure
-        # a UTF8 locale on Cirrus (or until the Gradle bug is fixed).
-        # TODO(amirh): Set the locale to UTF8.
-        - echo "$CIRRUS_CHANGE_MESSAGE" > /tmp/cirrus_change_message.txt
-        - echo "$CIRRUS_COMMIT_MESSAGE" > /tmp/cirrus_commit_message.txt
-        - export CIRRUS_CHANGE_MESSAGE=""
-        - export CIRRUS_COMMIT_MESSAGE=""
+        CPU: 2
+      script:
         - dart --enable-asserts ./dev/bots/test.dart
-        - export CIRRUS_CHANGE_MESSAGE=`cat /tmp/cirrus_change_message.txt`
-        - export CIRRUS_COMMIT_MESSAGE=`cat /tmp/cirrus_commit_message.txt`
+
+    - name: hostonly_devicelab_tests-misc-linux
       container:
-        cpu: 4
-        memory: 12G
-    - name: integration_tests_gradle2-linux
+        # Empirically, as of October 2019, the hostonly_devicelab_tests-misc-linux shard took about
+        # 20 minutes when run with 1 CPU and 4G of RAM, just over a quarter of an hour with 2 CPUs
+        # and 4G of RAM.
+        cpu: 2
       env:
-        SHARD: integration_tests
-        SUBSHARD: gradle2
-      test_script:
-        # Unsetting CIRRUS_CHANGE_MESSAGE and CIRRUS_COMMIT_MESSAGE as they
-        # might include non-ASCII characters which makes Gradle crash.
-        # See: https://github.com/flutter/flutter/issues/24935
-        # This is a temporary workaround until we figure how to properly configure
-        # a UTF8 locale on Cirrus (or until the Gradle bug is fixed).
-        # TODO(amirh): Set the locale to UTF8.
-        - echo "$CIRRUS_CHANGE_MESSAGE" > /tmp/cirrus_change_message.txt
-        - echo "$CIRRUS_COMMIT_MESSAGE" > /tmp/cirrus_commit_message.txt
-        - export CIRRUS_CHANGE_MESSAGE=""
-        - export CIRRUS_COMMIT_MESSAGE=""
+        CPU: 2
+      script:
         - dart --enable-asserts ./dev/bots/test.dart
-        - export CIRRUS_CHANGE_MESSAGE=`cat /tmp/cirrus_change_message.txt`
-        - export CIRRUS_COMMIT_MESSAGE=`cat /tmp/cirrus_commit_message.txt`
+
+    # TODO(ianh): name: add_to_app_tests-linux
+
+    - name: docs-linux # linux-only
+      only_if: "changesInclude('.cirrus.yml', 'dev/**', 'packages/flutter/**', 'packages/flutter_test/**', 'packages/flutter_drive/**', 'packages/flutter_localizations/**', 'packages/flutter_goldens/**', 'bin/internal/**') || $CIRRUS_PR == ''"
       container:
-        cpu: 4
-        memory: 12G
-    - name: release_smoke_tests
+        # Empirically, as of October 2019, the docs-linux shard took about 30 minutes when run with
+        # 1 CPU and 4G of RAM. 2 CPUs reduced that to 20 minutes, more CPUs did not improve matters.
+        cpu: 2
```

Comment:
On post submit, we do more work (we don't actually try to upload the docs on presubmit).  Have you tested post submits with this configuration to validate that it's acceptable?

Maybe at least bump these up when the `$CIRRUS_PR == ''`

Reply:
I don't want to increase the numbers unless we have reason to, as discussed above. If this fails on master I'll increase the numbers.